lib/object/validateDPObjectFactory.js

```diff
@@ -36,7 +39,13 @@ module.exports = function validateDPObjectFactory(
       return result;
     }
 
-    const enrichedDPContract = enrichDPContractWithBaseDPObject(dpContract);
+    if (!Object.prototype.hasOwnProperty.call(rawDPObject, '$action')) {
+      result.addError(
+        new MissingDPObjectActionError(rawDPObject),
+      );
+
+      return result;
+    }
 
     let dpObjectSchemaRef;
 
```

Comment:
We don't need `dpObjectSchemaRef` if action delete. You just need to check that object is defined (`isDPObjectDefined`).